bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,660 @@
+            if isinstance(s_ph.duration, str):
+                s_ph.duration = getattr(
+                    subphase_duration_params, s_ph.duration.replace("$", "")
+                )
+
+    def add_load_config(
+        self,
+        load: PowerCycleLoadConfig,
+        subphases: Union[str, Iterable[str]],
+        subphase_efficiency: Optional[List[Efficiency]] = None,
+    ):
+        """Add load config"""
+        self.loads[load.name] = load
+        if isinstance(subphases, str):
+            subphases = [subphases]
+        for subphase in subphases:
+            self.subphase[subphase].loads.append(load.name)
+            if subphase_efficiency is not None:
+                self.subphase[subphase].efficiencies[load.name] = subphase_efficiency
+
+    @classmethod
+    def from_json(cls, manager_config_path: Union[Path, str]):
+        """Create configuration from pure json"""
+        return cls.from_dict(read_json(manager_config_path))
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]):
+        """Create configuration from dictionary"""
+        return cls(
+            scenario=ScenarioConfig(**data["scenario"]),
+            pulse={
+                k: PulseConfig(name=k, **v) for k, v in data["pulse_library"].items()
+            },
+            phase={
+                k: PhaseConfig(name=k, **v) for k, v in data["phase_library"].items()
+            },
+            subphase={
+                k: PowerCycleSubPhase(name=k, **v)
+                for k, v in data["subphase_library"].items()
+            },
+            system={
+                k: PowerCycleSystem(name=k, **v)
+                for k, v in data["system_library"].items()
+            },
+            subsystem={
+                k: PowerCycleSubSystem(name=k, **v)
+                for k, v in data["sub_system_library"].items()
+            },
+            loads={
+                k: PowerCycleLoadConfig(name=k, **v)
+                for k, v in data["load_library"].items()
+            },
+        )
+
+    def make_phase(self, phase: str, *, check=True) -> Phase:
+        """Create a single phase object"""
+        if check:
+            self.check_config()
+
+        phase_config = self.phase[phase]
+        subphases = {k: self.subphase[k] for k in phase_config.subphases}
+        return Phase(
+            phase_config,
+            subphases,
+            Loads({
+                ld: self.loads[ld]
+                for subphase in subphases.values()
+                for ld in subphase.loads
+            }),
+        )
+
+    def make_pulse(self, pulse: str, *, check=True) -> Dict[str, Phase]:
+        """Create a pulse dictionary"""
+        if check:
+            self.check_config()
+        return {
+            phase: self.make_phase(phase, check=False)
+            for phase in self.pulse[pulse].phases
+        }
+
+    def make_scenario(self) -> Dict[str, Dict[str, Union[float, Dict[str, Phase]]]]:
+        """Create a scenario dictionary"""
```

Comment:
```suggestion
    def get_scenario(self) -> Dict[str, Dict[str, Union[float, Dict[str, Phase]]]]:
```

Also maybe break the return type up into type alliases?
It's pretty difficult to see where the float type comes from, i.e. either float or dict[str, phase]